x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -319,69 +318,115 @@ async function searchAvailableTasks({
+      OneOfTaskTypes('task.taskType', claimPartitions.unlimitedTypes),
+      // Either a task with idle status and runAt <= now or
+      // status running or claiming with a retryAt <= now.
+      shouldBeOneOf(IdleTaskWithExpiredRunAt, RunningOrClaimingTaskWithExpiredRetryAt),
+      // must have a status that isn't 'unrecognized'
+      RecognizedTask
+    );
+
+    const queryUnlimitedTasks = matchesClauses(
+      queryForUnlimitedTasks,
+      filterDownBy(InactiveTasks),
+      tasksWithPartitions(partitions)
+    );
+    searches.push({
+      query: queryUnlimitedTasks,
+      sort,
+      size,
+      seq_no_primary_term: true,
+    });
+  }
 
-  return await taskStore.fetch(
-    {
+  // add searches for limited types
+  for (const [type, capacity] of claimPartitions.limitedTypes) {
+    const queryForLimitedTasks = mustBeAllOf(
+      // Task must be enabled
+      EnabledTask,
+      // Specific task type
+      OneOfTaskTypes('task.taskType', [type]),
+      // Either a task with idle status and runAt <= now or
+      // status running or claiming with a retryAt <= now.
+      shouldBeOneOf(IdleTaskWithExpiredRunAt, RunningOrClaimingTaskWithExpiredRetryAt),
+      // must have a status that isn't 'unrecognized'
+      RecognizedTask
+    );
+
+    const query = matchesClauses(queryForLimitedTasks, filterDownBy(InactiveTasks));
+    searches.push({
       query,
       sort,
-      size,
+      size: capacity,
       seq_no_primary_term: true,
-    },
-    // limit the response size
-    true
-  );
+    });
+  }
+
+  return await taskStore.msearch(searches);
 }
 
-function applyLimitedConcurrency(
-  tasks: ConcreteTaskInstance[],
-  batches: TaskClaimingBatches
-): ConcreteTaskInstance[] {
-  // create a map of task type - concurrency
-  const limitedBatches = batches.filter(isLimited);
-  const limitedMap = new Map<string, number>();
-  for (const limitedBatch of limitedBatches) {
-    const { tasksTypes, concurrency } = limitedBatch;
-    limitedMap.set(tasksTypes, concurrency);
-  }
+interface ClaimPartitions {
+  removedTypes: string[];
+  unlimitedTypes: string[];
+  limitedTypes: Map<string, number>;
+}
 
-  // apply the limited concurrency
-  const result: ConcreteTaskInstance[] = [];
-  for (const task of tasks) {
-    const concurrency = limitedMap.get(task.taskType);
-    if (concurrency == null) {
-      result.push(task);
+interface BuildClaimPartitionsOpts {
+  types: Set<string>;
+  excludedTypes: Set<string>;
+  removedTypes: Set<string>;
+  getCapacity: (taskType?: string) => number;
+  definitions: TaskTypeDictionary;
+}
+
+function buildClaimPartitions(opts: BuildClaimPartitionsOpts): ClaimPartitions {
+  const result: ClaimPartitions = {
+    removedTypes: [],
+    unlimitedTypes: [],
+    limitedTypes: new Map(),
+  };
+
+  const { types, excludedTypes, removedTypes, getCapacity, definitions } = opts;
+  for (const type of types) {
+    if (excludedTypes.has(type)) continue;
```

Comment:
I noticed while adding an integration test https://github.com/elastic/kibana/pull/189431 that this uses slightly different logic than the default task claimer and doesn't respect wildcards. I think we should use the same function used for the default task claimer. Updated in my integration test PR so one of us will have a conflict!

Reply:
I ended up fixing this in the last main merge, since that was part of what the merge conflicted with.